lib/creator/yeoman/webpack-generator.js

```diff
@@ -1,138 +1,92 @@
-			console: false,
-			global: true,
-			process: true,
-			Buffer: true,
-			__filename: 'mock',
-			__dirname: 'mock',
-			setImmediate: true
-		}
-		this.configuration.dev.webpackOptions.performance = {
-			hints: "'warning'",
-			maxEntrypointSize: 400000,
-			maxAssetSize: 100000,
-			assetFilter: assetFilterFunction('js')
-		}
-		this.configuration.dev.webpackOptions.stats = 'errors-only'
-		this.configuration.dev.webpackOptions.amd = {
-			jQuery: true,
-			kQuery: false
-		}
-		this.configuration.dev.webpackOptions.bail = true;
-		this.configuration.dev.webpackOptions.cache = 'myCache'
-		this.configuration.dev.webpackOptions.profile = true
-		this.configuration.dev.webpackOptions.module = {
-			noParse: new RegExp(/jquery|lodash/),
-			rules: [
-				{
-					test: new RegExp('/\.jsx?$/'),
-					include: [
-						/* eslint-disable */
-						'path.resolve(__dirname, "app")'
-						/* eslint-enable */
-					],
-					exclude: [
-						/* eslint-disable */
-						'path.resolve(__dirname, ".." , "app", "demo-files")'
-						/* eslint-enable */
-					],
-					enforce: "'pre'",
-					loader: "'babel-loader'",
-					options: {
-						presets: ['es2015']
-					},
-				},
-				{
-					test: new RegExp('\\.html$'),
-					use: [
-						"'htmllint-loader'",
-						{
-							loader: "'html-loader'",
-							options: {
-								hello: "'world'"
+		let done = this.async();
+		let self = this;
+
+		this.configuration.config.webpackOptions.module = getDefaultLoaders();
+		this.configuration.config.webpackOptions.plugins = getDefaultPlugins();
+		this.prompt([
+			RawList('entryType', 'What kind of entry point do you want?', ['Array', 'Function', 'String', 'Object'])
+		]).then( (entryTypeAnswer) => {
+			// Ask different questions for entry points
+			entryQuestions(self, entryTypeAnswer).then(entryOptions => {
+				this.configuration.config.webpackOptions.entry = entryOptions;
+			}).then( () => {
+
+				this.prompt([
+					CheckList('outputType', 'What kind of output properties do you want?', [
+						'chunkFilename',
+						'filename',
+						'path'
+					])
+				]).then( (outputTypeAnswer) => {
+					// Ask different questions to output
+					OutputQuestions(self, outputTypeAnswer['outputType']).then(outputOptions => {
+						this.configuration.config.webpackOptions.output = outputOptions;
+					}).then( () => {
+						// Ask if the user wants to use extractPlugin
+						this.prompt([
+							Input('extractPlugin', 'If you got a css file, what\'s it named?')
+						]).then( (extractAnswer) => {
+							if(extractAnswer['extractPlugin'].length !== 0) {
+								this.configuration.config.webpackOptions.plugins.push(
+									'new ExtractTextPlugin(\'styles.css\')'
+								);
+								this.configuration.config.webpackOptions.module.rules.push({
+									test: new RegExp(/\.css$/),
+									use: `ExtractTextPlugin.extract({
+										fallback: 'style-loader',
+										use: 'css-loader'
+									})`
+								});
```

Comment:
Could be overkill as well as another callback that needs to be added. I'm really favoring rxJS w/ inquirer for prompting, cause the amount of callbacks are unbelievable with regular syntax. I'm open to the suggestion, this would inject the webpack bundle for them, right? If so, definitely should be added.